cmd/zc_traverser_local.go

```diff
@@ -325,11 +325,14 @@ func cleanLocalPath(localPath string) string {
 	// return normalizedPath path separator.
 	normalizedPath = strings.ReplaceAll(normalizedPath, common.AZCOPY_PATH_SEPARATOR_STRING, localPathSeparator)
 
-	if strings.HasPrefix(localPath, `\\`) || strings.HasPrefix(localPath, `//`) { // path.Clean steals the first / from the // or \\ prefix.
-		// return the \ we stole from the UNC path.
+	// path.Clean steals the first / from the // or \\ prefix.
+	if strings.HasPrefix(localPath, `\\`) || strings.HasPrefix(localPath, `//`) {
+		// return the \ we stole from the UNC/extended path.
 		normalizedPath = localPathSeparator + normalizedPath
-	} else if len(localPath) == 3 && (strings.HasSuffix(localPath, `:\`) || strings.HasSuffix(localPath, ":/")) ||
-		len(localPath) == 2 && strings.HasSuffix(localPath, ":") { // path.Clean steals the last / from C:\, C:/, and does not add one for C:
+	}
+
+	// path.Clean steals the last / from C:\, C:/, and does not add one for C:
+	if common.RootDriveRegex.MatchString(strings.ReplaceAll(common.ToShortPath(normalizedPath), common.OS_PATH_SEPARATOR, common.AZCOPY_PATH_SEPARATOR_STRING)) {
```

Comment:
This looks like a bug.  Because its testing normalizedPath, which will have already have the / stripped, against a regex that expects the / to be present.  Do you need a different regex?

Reply:
Or maybe test localPath against the regex you already have, instead of testing normalized path.

Reply:
The regex does not expect the / to be present.

Please check the regexr example provided in the comment above the regex.
https://regexr.com/4mf6l

Reply:
Oops I missed the ? in the regex.